src/models/BaseModel.py

```diff
@@ -1,118 +1,190 @@
-        """
-        :param train_size: can be either a float or int
-            - float: ratio of how much is training/test data
-            - int: for total size
-        :param test_size: see train_size
-        :param load_location: boolean, true if we want to load and process the normalized location
-        """
-        if BaseModel.data is None:
-            self.load_csv_data(train_data_csv_file_name)
-
-        # split original training dataset into train and test data
-        # since we most likely wont get the test salary
-        self.train_data, self.test_data = train_test_split(BaseModel.data,
-                                                           train_size=train_size,
-                                                           test_size=test_size)
-        self.training_data_size = len(self.train_data)
-        self.test_data_size = len(self.test_data)
-
-        self.train_description_feature = self.train_data[['FullDescription']]
-        self.train_contract_type_feature = self.train_data[['ContractType']]
-        self.train_contract_time_feature = self.train_data[['ContractTime']]
-        self.train_category_feature = self.train_data[['Category']]
-        self.train_company_feature = self.train_data[['Company']]
-        self.train_source_name_feature = self.train_data[['SourceName']]
-        self.train_location_raw_feature = self.train_data[['LocationRaw']]
-        self.train_salary_normalized = self.train_data[['SalaryNormalized']]
-
-        self.test_description_feature = self.test_data[['FullDescription']]
-        self.test_salary_normalized = self.test_data[['SalaryNormalized']]
-
-        # can be disabled to speed up load times
-        if load_location:
-            if BaseModel.normalized_location_data is None:
-                self.load_location_data(train_normalized_location_file_name)
-            train_normalized_location_data = \
-                BaseModel.normalized_location_data[:self.training_data_size]
-
-            # test_normalized_location_data = normalized_location_data[-self.test_data_size:]
-
-            self.cleaned_town_feature = train_normalized_location_data[['town']]
-            self.cleaned_region_feature = train_normalized_location_data[['region']]
-
-    @staticmethod
-    def load_location_data(train_normalized_location_file_name):
-        BaseModel.normalized_location_data = pd.read_csv(train_normalized_location_file_name)
+    TRAIN_NORMALIZED_LOCATION_FILE_NAME = '../data/train_normalised_location.csv'
+    TRAIN_DATA_CSV_FILE_NAME = '../data/Train_rev1.csv'
+    CLEANED_FILE_NAME = '../data/Preprocessed_Data.csv'
+
+    def __init__(self, train_size=0.75):
+        print('Initialized BaseModel')
+        if os.path.exists(BaseModel.CLEANED_FILE_NAME):
+            print('Pre-processed data exists, reading from the file')
+            self.cleaned_encoded_data = self.load_cleaned_data()
+            print('Data read complete')
+        else:
```

Comment:
If processed file is not available, this is where all the cleaning and exporting happens
(and in functions such as export_data | preprocess_data )